solr/core/src/java/org/apache/solr/handler/admin/InfoHandler.java

```diff
@@ -50,7 +50,10 @@ public InfoHandler(final CoreContainer coreContainer) {
     handlers.put("properties", new PropertiesRequestHandler());
     handlers.put("logging", new LoggingHandler(coreContainer));
     handlers.put("system", new SystemInfoHandler(coreContainer));
-    handlers.put("health", new HealthCheckHandler(coreContainer));
+    if (coreContainer.getHealthCheckHandler() == null) {
+      throw new IllegalStateException("HealthCheckHandler needs to be initialized before creating InfoHandler");
```

Comment:
> `throw new IllegalStateException("HealthCheckHandler needs to be initialized before creating InfoHandler");`

How about also having a "HealthCheckHandler then InfoHandler order matters because ..." style comment in `CoreContainer`?